### PR TITLE
fix for belongs_to relation

### PR DESCRIPTION
**Pull request by isaacwecode**
- fixed bug with dollar sign in belongs_to
- belongs_to now supports dot notation
- fixed broken test

### CODE REVIEW

**Comment by Firesphere** [comment]
You don't think, pushing to a new function is a bit overkill? Abstracting to a level where you're just pointing to other functions for the sake of abstracting is not really a good idea.

**Comment by isaacwecode** [comment]
You would prefer two times the exact same explode statement in two places?

**Comment by axyr** [comment]
the extra method is ok...
but travis fails...

**Comment by isaacwecode** [comment]
Travis is getting different results from the tests than I am when I run it locally.
I'll revert my test fix and it should be OK

**Comment by axyr** [comment]
could you post your failed local test output?

File: code/Generators/OrmTagGenerator.php [hunk 1]
```diff
@@ -92,7 +92,8 @@ protected function generateBelongsToTags()
     {
         if ($fields = (array)$this->getClassConfig('belongs_to')) {
             foreach ($fields as $fieldName => $dataObjectName) {
-                $this->pushMethodTag($fieldName, $dataObjectName . " \$$fieldName");
```
Comment (axyr) [inline code/Generators/OrmTagGenerator.php]: nice catch. will try to find a test for that, since it passed anyway..

**Comment by isaacwecode** [comment]
ALL TESTS
Running test cases: 
--------------------
annotatechangeddbspecstest, annotateclassinfotest, annotatepermissioncheckertest,
controllerannotatortest, dataobjectannotatortest

................F
DataObjectAnnotatorTest::testAnnotateDataExtension
Failed asserting that '<?php

/**
 * Class DataObjectAnnotatorTest_Team_Extension
 *
 * @property DataObjectAnnotatorTest_Team|DataObjectAnnotatorTest_TeamChanged $owner
 * @property string $ExtendedVarcharField
 * @property int $ExtendedIntField
 * @property int $ExtendedHasOneRelationshipID
 * @method DataObjectTest_Player ExtendedHasOneRelationship()
 */
class DataObjectAnnotatorTest_Team_Extension extends DataExtension implements TestOnly
{

    private static $db = array(
        'ExtendedVarcharField' => 'Varchar',
        'ExtendedIntField'     => 'Int'
    );

    private static $has_one = array(
        'ExtendedHasOneRelationship' => 'DataObjectTest_Player'
    );
}
' contains "@property DataObjectAnnotatorTest_Team $owner".
PHPUnit_Framework_Assert::assertContains(@property DataObjectAnnotatorTest_Team $owner,<?php

/**
 * Class DataObjectAnnotatorTest_Team_Extension
 *
 * @property DataObjectAnnotatorTest_Team|DataObjectAnnotatorTest_TeamChanged $owner
 * @property string $ExtendedVarcharField
 * @property int $ExtendedIntField
 * @property int $ExtendedHasOneRelationshipID
 * @method DataObjectTest_Player ExtendedHasOneRelationship()
 */
class DataObjectAnnotatorTest_Team_Extension extends DataExtension implements TestOnly
{

    private static $db = array(
        'ExtendedVarcharField' => 'Varchar',
        'ExtendedIntField'     => 'Int'
    );

    private static $has_one = array(
        'ExtendedHasOneRelationship' => 'DataObjectTest_Player'
    );
}
,,,1,)
SapphireTest.php:542

SapphireTest::assertContains(@property DataObjectAnnotatorTest_Team $owner,<?php

/**
 * Class DataObjectAnnotatorTest_Team_Extension
 *
 * @property DataObjectAnnotatorTest_Team|DataObjectAnnotatorTest_TeamChanged $owner
 * @property string $ExtendedVarcharField
 * @property int $ExtendedIntField
 * @property int $ExtendedHasOneRelationshipID
 * @method DataObjectTest_Player ExtendedHasOneRelationship()
 */
class DataObjectAnnotatorTest_Team_Extension extends DataExtension implements TestOnly
{

    private static $db = array(
        'ExtendedVarcharField' => 'Varchar',
        'ExtendedIntField'     => 'Int'
    );

    private static $has_one = array(
        'ExtendedHasOneRelationship' => 'DataObjectTest_Player'
    );
}
)
DataObjectAnnotatorTest.php:129

--------------------
..

 AT LEAST ONE FAILURE 

19 tests run: 18 passes, 1 failures, and 0 incomplete
Maximum memory usage: 89.6M

Total time: 2.446 seconds

**Comment by axyr** [comment]
Thanks. will take look at it later!

**Pull request merged**